v3/src/components/graph/adornments/count/count-model.ts

```diff
@@ -1,12 +1,43 @@
-import { Instance } from "@concord-consortium/mobx-state-tree"
+import { Instance, types } from "mobx-state-tree"
 import { AdornmentModel, IAdornmentModel } from "../adornment-models"
 import { kCountType } from "./count-types"
 
 export const CountModel = AdornmentModel
   .named('CountModel')
   .props({
-    type: 'Count'
+    type: 'Count',
+    showCount: false,
+    showPercent: false,
+    percentType: types.optional(types.enumeration(["cell", "column", "row"]), "row"),
   })
+  .views(self => ({
+    percentValue(dataConfig: any, casesInPlot: number, cellKey: Record<string, string>) {
+      const casesInRow = dataConfig?.rowCases(cellKey)?.length ?? 0
+      const casesInColumn = dataConfig?.columnCases(cellKey)?.length ?? 0
+      const casesInCell = dataConfig?.cellCases(cellKey)?.length ?? 0
+      const divisor = self.percentType === "row"
+        ? casesInRow
+        : self.percentType === "column"
+          ? casesInColumn
+          : casesInCell
+      const percentValue = ((casesInPlot / divisor) * 100).toFixed(2)
+      return isFinite(Number(percentValue)) ? Math.round(Number(percentValue) * 100)/100 : 0
```

Comment:
Some unnecessary conversions going on here. How about something like:
```suggestion
      const percentValue = casesInPlot / divisor
      return isFinite(percentValue) ? percentValue : 0
```
and then leave it to the view to do any necessary rounding for display purposes. The model should return the actual percent value.

Reply:
@kswenson Understood about leaving it to the view to do the necessary rounding. That makes a lot of sense.

To get the correct display value, though, I think at least some of the conversion was actually necessary. The way I was doing it definitely wasn't optimal or clear, though. I made the changes you suggested here, but also added a `percentString` utility function that handles the formatting in a clearer way. It's used by the Count component to set the value that gets displayed. Does that look good? Or would it be better to make `percentString` a view on the count model?